charts/helm-controller-operator/templates/helmController/service.yaml

```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "helm-controller-operator.fullname" . }}-metrics-service
+  labels:
+    {{- include "helm-controller-operator.labels" . | nindent 4 }}
+spec:
+  ports:
+    {{- if .Values.kubeRBACProxy.enabled }}
+    - name: https-metrics
+      port: 8443
+      targetPort: https-metrics
+      protocol: TCP
+    {{- else }}
+    - name: http-metrics
+      port: 8080
+      targetPort: http-metrics
+      protocol: TCP
+    {{- end }}
+  selector:
+    {{- include "helm-controller-operator.selectorLabels" . | nindent 4 }}
```

Comment:
When you describe this service do you see all its pods in the status or pods from the sourceController as well ?
Hmm I think you should have different selector labels for the two deployment, e.g. by adding `app.kubernetes.io/component: helm-controller` to labels and selectors (e.g. as the [chart we saw does](https://github.com/grafana/helm-charts/blob/main/charts/tempo-distributed/templates/querier/_helpers-querier.tpl))